File: drug-discovery-ui/src/services/interactions.ts
export type InteractionPair = {
  drugs: string[]
  type: 'synergistic' | 'antagonistic' | 'additive' | 'none'
  severity: 'mild' | 'moderate' | 'severe' | 'contraindicated'
  mechanism: string
  clinicalSignificance: string
  recommendations: string
}

export type InteractionResponse = {
  overallSafety: 'safe' | 'caution' | 'dangerous' | 'contraindicated'
  interactions: InteractionPair[]
  saferAlternatives: string[]
  heuristic?: boolean
  error?: string
}

export async function analyzeInteractions(drugs: string[]): Promise<InteractionResponse> {
  const resp = await fetch('/api/v1/interactions/analyze', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ drugs }),
  })
  if (!resp.ok) {
    const txt = await resp.text()
    throw new Error(`Server error: ${resp.status} ${txt}`)
  }
  return (await resp.json()) as InteractionResponse
}

File: drug-discovery-ui/backend/app/api/models/interactions.py
from typing import List, Literal, Optional
from pydantic import BaseModel, Field

Severity = Literal['mild', 'moderate', 'severe', 'contraindicated']
InteractionType = Literal['synergistic', 'antagonistic', 'additive', 'none']
OverallSafety = Literal['safe', 'caution', 'dangerous', 'contraindicated']

class InteractionPair(BaseModel):
    drugs: List[str] = Field(..., min_items=2)
    type: InteractionType
    severity: Severity
    mechanism: str
    clinicalSignificance: str
    recommendations: str

class InteractionRequest(BaseModel):
    drugs: List[str] = Field(..., min_items=2, max_items=5)

class InteractionResponse(BaseModel):
    overallSafety: OverallSafety
    interactions: List[InteractionPair]
    saferAlternatives: List[str] = []
    heuristic: bool = False
    error: Optional[str] = None
